app/src/app/components/contents/common/FilterByName.tsx

```diff
@@ -0,0 +1,53 @@
+import debounce from "lodash.debounce";
+import { Dispatch, SetStateAction, useCallback, useEffect, useRef } from "react";
+import { Input } from "../../Base/Input";
+import { Button } from "../../Base/Button";
+import { X } from "lucide-react";
+
+interface FilterByNameProps {
+  setFilterByName: Dispatch<SetStateAction<string>>;
+  setPageNumber?: Dispatch<SetStateAction<number>>;
+}
+export const FilterByName = ({ setFilterByName, setPageNumber }: FilterByNameProps) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleSetNameFilter = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setFilterByName(event.target.value);
+    if (setPageNumber) {
+      setPageNumber(0);
```

Comment:
OK, so when the filter changes it resets the page number to 0. I'm not sure it's really the responsibility of the filter component to worry about paging. Perhaps the paging component should do the reset when its item list changes, But don't feel too strongly about that.

Reply:
yup fully agreee with that too.. have now passed in postfilter action that parent is responsible for